### ISSUE TITLE
Provide a converter for detekt

### ISSUE THREAD

**Issue by lgolding** [microsoft/sarif-sdk#2091]
https://github.com/detekt/detekt

@michaelcfanning per request from @matt-desai.

**Comment by eddynaka** [comment]
cc: @yongyan-gh

**Comment by yongyan-gh** [comment]
detekt implemented its own SARIF export @ https://github.com/detekt/detekt/issues/3045

**Comment by mattdesai** [comment]
yes, we are testing v1.16-rc as we speak.  hopefully, it looks good and we can get a final drop of 1.16 soon.

**Comment by chao2zhang** [comment]
It looks like detekt 1.16 release has a bug that does not report any `ReportingDescriptor` for `rules`. I am going to verify if Github action still works without it, and hopefully we can make a patch version of Detekt soon.
